azure-pipelines.yml

```diff
@@ -1,162 +1,888 @@
+                  targetFolder: '$(Build.ArtifactStagingDirectory)\CDNFeed\$(package.version)\js'
+                displayName: 'Copy TeamsJS Content for CDN'
+              - task: CopyFiles@2
+                inputs:
+                  Contents: |
+                    packages\teams-js\package.json
+                    packages\teams-js\README.md
+                    LICENSE
+                  TargetFolder: '$(Build.ArtifactStagingDirectory)\NPMFeed'
+                  flattenFolders: true
+                displayName: 'Copy TeamsJS Content for NPM'
+              - task: CopyFiles@2
+                inputs:
+                  Contents: |
+                    packages\teams-js\dist\**\?(*.js|*.ts|*.map)
+                  TargetFolder: '$(Build.ArtifactStagingDirectory)\NPMFeed\dist'
+                  flattenFolders: true
+                displayName: 'Copy JS Content for NPM'
+              - task: CopyFiles@2
+                inputs:
+                  Contents: |
+                    packages\teams-js\scripts\*.ps1
+                  TargetFolder: '$(Build.ArtifactStagingDirectory)\scripts'
+                  flattenFolders: true
+                displayName: 'Copy Power Shell scripts to scripts'
+            templateContext:
+              outputs:
+                - output: pipelineArtifact
+                  displayName: 'Publish bundle analysis'
+                  condition: and( in(variables['Agent.JobStatus'], 'Succeeded', 'SucceededWithIssues'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                  targetPath: './common/temp/bundleAnalysis'
+                  artifactName: '$(bundleArtifactName)'
+                - output: pipelineArtifact
+                  displayName: 'Publish Test app artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\teams-test-app\$(Build.BuildId).zip'
+                  artifactName: 'teams-test-app'
+                - output: pipelineArtifact
+                  displayName: 'Publish CDN feed to build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\CDNFeed'
+                  artifactName: 'CDNFeed'
+                - output: pipelineArtifact
+                  displayName: 'Publish NPM feed to Build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\NPMFeed'
+                  artifactName: 'NPMFeed'
+                - output: pipelineArtifact
+                  displayName: 'Publish Powershell Scripts to Build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\scripts'
+                  artifactName: 'scripts'
+                - output: pipelineArtifact
+                  displayName: 'Publish SBOM to Build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\_manifest'
+                  artifactName: 'SBOM'
+          - job: E2ETest1
+            displayName: 'E2E Test - Perf'
+            pool:
```

Comment:
All of these nested pools were added by me by hand after the tool ran because the tool tried to convert our entire pipeline to run on windows-latest. I don't know if this will work or not!